### PR TITLE
modem: Disable trace_proxy_irq_handler when trace is disabled

### PR DESCRIPTION
**Pull request by SebastianBoe**
trace_proxy_irq_handler is using an IRQ that other users also want to
use. For instance the test zephyr/tests/kernel/gen_isr_table/ wants to
use this IRQ.

I don't see why we need to enable the IRQ when trace is disabled. So
this patch disables it when NRF_MODEM_LIB_TRACE_ENABLED=n.

NCSDK-11344

Signed-off-by: Sebastian Bøe <sebastian.boe@nordicsemi.no>

### CODE REVIEW

File: lib/nrf_modem_lib/nrf_modem_os.c [hunk 1]
```diff
@@ -509,11 +511,9 @@ void trace_uart_init(void)
 #define RTT_BUF_SZ		(CONFIG_NRF_MODEM_LIB_TRACE_MEDIUM_RTT_BUF_SIZE)
 static int trace_rtt_channel;
 static char rtt_buffer[RTT_BUF_SZ];
-#endif
```
Comment (rlubos) [inline lib/nrf_modem_lib/nrf_modem_os.c:512]: That way, `trace_rtt_init()` will no longer be available if `NRF_MODEM_LIB_TRACE_ENABLED=y` and UART is used for the medium. Would need more sophisticated ifdef slicing when calling initializers (it's either UART or RTT as a medium choice, never both).

Comment (SebastianBoe) [reply lib/nrf_modem_lib/nrf_modem_os.c:512]: Good catch.

Fixed.

**Comment by SebastianBoe** [comment]
This PR assumes that trace_uart_init and friends are accidentally non-static.

**Review by rlubos — APPROVED** [review]

**Review by lemrey — COMMENTED** [review]
Looks good, you could prefix it `lib: nrf_modem_lib: ...` ?

**Comment by SebastianBoe** [comment]
@lemrey : Done

**Comment by SebastianBoe** [comment]
@lemrey : Could I get a +1 ?

**Comment by lemrey** [comment]
@SebastianBoe did you forget to push the most recent commit?

**Comment by SebastianBoe** [comment]
oops

**Comment by SebastianBoe** [comment]
@lemrey : Fixed

**Review by lemrey — APPROVED** [review]

**Pull request merged**